### DISCUSSION TITLE
select by name

### DISCUSSION THREAD

**Discussion by ASamSam** [eclipse-sirius/sirius-desktop#405] [Q&A] [answered]
Hello. can you please tell me, what am I doing wrong?
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/29237978/d90288a9-464b-4268-9400-49d051e29674)
I do perform request like 'aql:self.target.produces->select(f | f.name <> 'dd')'
and I have object named dd in 'produced' list (and only that object, no others).
 that is exactly what I see in output, but I asked to get objects not named as 'dd' ....

**Comment by lredor** [comment]
Hi,
No idea. What is the warning in the Sirius interpreter view ?
A similar expression is OK on my side:
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/1825882/a7fa41cd-bd9d-48b2-b307-85173d98787c)

**Comment by ASamSam** [comment]
Hello. Thank you for the answer. It seems I was confused between 'name' and 'Name'. I have attribute 'Name' in 'Data' class, and did expect to access it.
i would say, this is confusing for me:
aql:self.name : 'dd'
aql:self.Name : warning

aql:self.target.name : warning
aql:self.target.Name : 'dd'

I thought, AQL is case-insensitive...

**Reply by lredor** [reply, marked as answer]
- `aql:self.name` returns the name of the [`DNode`](https://github.com/eclipse-sirius/sirius-desktop/blob/master/plugins/org.eclipse.sirius.diagram.model/src-gen/org/eclipse/sirius/diagram/impl/DNodeImpl.java) (more precisely of the [`DRepresentationElement`](https://github.com/eclipse-sirius/sirius-desktop/blob/master/plugins/org.eclipse.sirius.model/src-gen/org/eclipse/sirius/viewpoint/impl/DRepresentationElementImpl.java#L78)). The DNode is the graphical representation of your semantic element.
- `aql:self.target.Name` returns the name of your semantic element.

**Reply by ASamSam** [reply]
yes, thank you very much!
DRepresentationElement has the name of the referenced object because of my config. ok
